apps/opencs/view/render/terraintextureselection.cpp

```diff
@@ -0,0 +1,109 @@
+#include "terraintextureselection.hpp"
+
+#include <osg/Geode>
+#include <osg/Geometry>
+#include <osg/PositionAttitudeTransform>
+
+CSVRender::TerrainTextureSelection::TerrainTextureSelection(osg::Group* parentNode, const CSMWorld::CellCoordinates& coords, const ESM::Land& esmLand)
+    :TerrainSelection {coords, esmLand, parentNode}
+{
+    osg::ref_ptr<osg::Geometry> geometry {new osg::Geometry{}};
+
+    // Color
+    osg::ref_ptr<osg::Vec4Array> colors {new osg::Vec4Array{}};
+    colors->push_back(osg::Vec4f(0.f, 0.5f, 0.f, 1.f));
+
+    geometry->setColorArray(colors, osg::Array::Binding::BIND_OVERALL);
+
+    // Primitive
+    osg::ref_ptr<osg::DrawArrays> drawArrays {new osg::DrawArrays{osg::PrimitiveSet::Mode::LINES}};
+
+    geometry->addPrimitiveSet(drawArrays);
+    geometry->getOrCreateStateSet()->setMode(GL_LIGHTING, osg::StateAttribute::Values::OFF);
+
+    mGeode = new osg::Geode{};
+    mGeode->addDrawable(geometry);
+    getBaseNode()->addChild(mGeode);
+}
+
+void CSVRender::TerrainTextureSelection::addToSelection(osg::Vec3d worldPos)
+{
+    const TerrainPos localPos {toTextureCoords(worldPos)};
+
+    if (std::find(mSelection.begin(), mSelection.end(), localPos) == mSelection.end())
```

Comment:
Two of the travis builds are failing here with 
/home/travis/build/OpenMW/openmw/apps/opencs/view/render/terraintextureselection.cpp:33:65: error: no matching function for call to ‘find(std::vector<std::pair<int, int> >::iterator, std::vector<std::pair<int, int> >::iterator, const TerrainPos&)’